portability-transfer/src/main/java/org/datatransferproject/transfer/copier/PortabilityAbstractInMemoryDataCopier.java

```diff
@@ -125,6 +126,8 @@ protected ExportResult<?> copyIteration(
     DataModel exportedData = exportResult.getExportedData();
     if (exportedData != null) {
       PortabilityJob job = jobStore.findJob(jobId);
+      Preconditions.checkNotNull(job, "Transfer cannot proceed without fetching the job. "
```

Comment:
lgtm; optional nit about including id and compacting the message a bit:

```suggestion
      checkNotNull(job, "Transfer job somehow from jobStore, mid-way through copy iteration, for the Job ID \"%s\"", jobId.toString());
```

Reply:
Done